### ISSUE TITLE
[req]: Disable Raw Search for EZTV Indexer

### ISSUE THREAD

**Issue by alphabitnz** [Jackett/Jackett#13469]
### Is there already a request for your feature?

- [X] I have checked older issues, open and closed

### Is your feature request related to a problem? Please describe.

Torznab search via eztv indexer fails to return results when %27 (apostrophe) character included in URL
e.g. Shaun%20Micallef%27s%20MAD%20AS%20HELL
The same search successfully returns results when %27 (apostrophe) is omitted
e.g. Shaun%20Micallefs%20MAD%20AS%20HELL

### Describe the solution you'd like

Disable Raw Search for the built-in EZTV indexer

**Comment by garfield69** [comment]
@bakerboy448 you brought this across to Jackett, does Prowlerr have this problem, or is Jackett missing some additional support code in cardigann or whatever?

**Comment by bakerboy448** [comment]
Was added 12/2021 and not a single issue has been reported since.

https://github.com/Prowlarr/Indexers/issues/78

And a search on EZTV's series page indicates they do not have any series under the title Shaun Micallef and thus there seems to be no issue here.

Raw Search MUST be enabled.  If apostrophes or quotes need to be removed then that needs to be done keyword side.

If raw search is removed then \*arrs will send `Grey s Anatomy` rather than `Grey's Anatomy` and searching with the space returns no results, so it cannot be removed.

It's possible it needs a keyword replace added, but I don't believe that can nor should be concluded by a series that does not exist in their series list.

No action to take based on this GHI alone given EZTV only supports specific series that are on their site.

**Comment by ilike2burnthing** [comment]
Apostrophes are only an issue in so far as most release groups remove them from titles, and so searching with it will return no results.

If you search for one of the few shows where it is used by at least a few groups, e.g. `lawrence o'donnell`, then you'll see the site has no issue with apostrophes.

Searching for Grey's Anatomy S18E20 through Sonarr results in `Grey%27s-Anatomy-S18E20` which returns no results.

Disabling `allowrawsearch` and searching again results in `Greys-Anatomy-S18E20` which returns 8 results.

**Comment by bakerboy448** [comment]
\*arrs will not send greys anatomy if RawSearch is removed. so that test is invalid

**Comment by ilike2burnthing** [comment]
Yep, edited the .yml to comment out `allowrawsearch`, restarted Jackett then Sonarr, enabled enhanced logging in Jackett, performed a manual search for Grey's Anatomy S18E20 in Sonarr, it found 8 results, and Jackett's log showed the URL as https://eztv.re/search/Greys-Anatomy-S18E20

**Comment by bakerboy448** [comment]
There's no reason that sonarr would be sending `greys` to jackett with raw search removed as my understanding.

It's Always Sunny is what triggered the initial addition.
https://github.com/Prowlarr/Indexers/issues/78

**Comment by alphabitnz** [comment]
> And a search on EZTV's series page indicates they do not have any series under the title Shaun Micallef and thus there seems to be no issue here.

That's because you need to search for title with "Shaun Micallefs"

**Comment by alphabitnz** [comment]
Current behaviour for TPB indexer works correctly for this example show, whereas EZTV indexer doesn't.
This from Sonarr debug log:

`2022-08-14 13:32:31.3|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=Shaun%20Micallef%27s%20MAD%20AS%20HELL&season=14&ep=12
2022-08-14 13:32:31.3|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/thepiratebay/results/torznab/api?t=tvsearch&cat=5000,5040,5050,100205,100208&extended=1&apikey=(removed)&offset=0&limit=100&q=Shaun%20Micallefs%20MAD%20AS%20HELL&season=14&ep=12`

**Comment by alphabitnz** [comment]
> It's Always Sunny is what triggered the initial addition.

Try searching now at eztv dot re for "always sunny" and you'll see "Its" not It's" in the results - possible that this has changed recently if that was not previously the case in Dec '21

**Comment by bakerboy448** [comment]
> > It's Always Sunny is what triggered the initial addition.
> 
> 
> 
> Try searching now at eztv dot re for "always sunny" and you'll see "Its" not "It's" in the results - possible that this has changed recently if that was not previously the case in Dec '21

You didn't look at the referenced issue at all?

> tested eztv looks like they need to be changed to support rawsearch so it's becomes its rather than it s

**Comment by bakerboy448** [comment]
Got clarification from Markus on what's sent

> usually `'` is replaced with nothing, not a space

(For when there isn't raw search)

So if Eztv needs `9-1-1` or `911` rather than `9 1 1` then we need RawSearch to stay and a keywordfilters accordingly

`'` => `""` for starters

**Comment by ilike2burnthing** [comment]
Retested in Windows Sandbox, this time making changes to the .yml file before even installing Sonarr.

---

Sonarr debug log for search of `Grey's Anatomy S18E20`:

```text
2022-08-14 16:50:48.5|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=Greys%20Anatomy&season=18&ep=20
2022-08-14 16:50:50.1|Debug|ReleaseSearchService|Total of 8 reports were found for [Grey's Anatomy : S18E20] from 1 indexers
```

Jackett uses URL - https://eztv.re/search/Greys-Anatomy-S18E20

---

Sonarr debug log for search of `It's Always Sunny in Philadelphia S15E08`:

```text
2022-08-14 17:00:55.4|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=Its%20Always%20Sunny%20in%20Philadelphia&season=15&ep=8
2022-08-14 17:00:56.3|Debug|ReleaseSearchService|Total of 11 reports were found for [It's Always Sunny in Philadelphia : S15E08] from 1 indexers
```

Jackett uses URL - https://eztv.re/search/Its-Always-Sunny-in-Philadelphia-S15E08

**Comment by alphabitnz** [comment]
> So if Eztv needs `9-1-1` or `911` rather than `9 1 1` (no raw search) then we need RawSearch to stay and a keywordfilters accordingly

If Eztv actually has show titles that require RawSearch then I agree the solution would be more complicated than just disabling RawSearch for that Indexer, but note the referenced issue https://github.com/Prowlarr/Indexers/issues/78 was saying that 1337x was failing to give results, and EZTV was working correctly at that time: "EZTV bring back all new releases".

**Comment by bakerboy448** [comment]
> EZTV bring back all new releases

Means every new upload. Not the expected results for the series searched.  Eztv does this rather than returning no results.

**Comment by alphabitnz** [comment]
> Eztv does this rather than returning no results.

If I pass an intentional fail query URL such as "Nonexist Show" S01E01:
`127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=Nonexist%20Show&season=1&ep=1`

The resulting XML file returns no results, not every new upload.

Curious to hear of any examples where Eztv requires RawSearch enabled, and whether you still see the mentioned behaviour about returning all new releases if you test currently.

**Comment by bakerboy448** [comment]
Jackett does additional filtering, so you cannot use jackett to look at the actual results returned

**Comment by alphabitnz** [comment]
Ok then. Sounds complicated but I'm sure you'll figure out the best way to solve the apostrophe problem.

**Comment by ilike2burnthing** [comment]
With `allowrawsearch` and the recently added keywordsfilters disabled:

```text
2022-08-14 17:19:40.7|Info|ReleaseSearchService|Searching indexers for [9-1-1 : S05E18]. 1 active indexers
2022-08-14 17:19:40.8|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=911&season=5&ep=18
2022-08-14 17:19:52.4|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=9&season=5&ep=18
2022-08-14 17:19:55.1|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=9%201%201&season=5&ep=18
2022-08-14 17:19:57.5|Debug|ReleaseSearchService|Total of 32 reports were found for [9-1-1 : S05E18] from 1 indexers
```

Jackett uses:
https://eztv.re/search/911%20S05E18
https://eztv.re/search/9%20S05E18
https://eztv.re/search/9%201%201%20S05E18

With the keywordsfilters that would be:
https://eztv.re/search/911-S05E18
https://eztv.re/search/9-S05E18
https://eztv.re/search/9-1-1-S05E18

---

With `allowrawsearch` enabled:

```text
2022-08-14 17:37:08.8|Info|ReleaseSearchService|Searching indexers for [9-1-1 : S05E18]. 1 active indexers
2022-08-14 17:37:08.8|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=911&season=5&ep=18
2022-08-14 17:37:10.0|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=9&season=5&ep=18
2022-08-14 17:37:13.9|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/api/v2.0/indexers/eztv/results/torznab/api?t=tvsearch&cat=5000&extended=1&apikey=(removed)&offset=0&limit=100&q=9-1-1&season=5&ep=18
2022-08-14 17:37:16.9|Debug|ReleaseSearchService|Total of 32 reports were found for [9-1-1 : S05E18] from 1 indexers
```

Jackett uses:
https://eztv.re/search/911%20S05E18
https://eztv.re/search/9%20S05E18
https://eztv.re/search/9-1-1%20S05E18

With the keywordsfilters that would be:
https://eztv.re/search/911-S05E18
https://eztv.re/search/9-S05E18
https://eztv.re/search/911-S05E18

---

With or without Raw Search, with the indexer's current keywordsfilters the same number of results are found. The only difference is that Raw Search duplicates a search.

Are there other scenarios in which Raw Search would generally be needed? If it's avoidable I'd rather not completely remove the ability for people to use apostrophes in manual Jackett searches for EZTV, as the trackers supports and uses them.

**Comment by bakerboy448** [comment]
Rip it out for now and can evaluate later and see how it goes

**Comment by garfield69** [comment]
v0.20.1698

**Comment by chrise86** [comment]
This seems to still be an issue in v0.20.1768

**Comment by alphabitnz** [comment]
> This seems to still be an issue in v0.20.1768

It's been fixed for me for a while now. Confirmed still fixed in my case, currently running v0.20.1796
